src/rmkit/fb/fb.cpy

```diff
@@ -329,7 +329,11 @@ namespace framebuffer:
             break
 
           if src[i] != alpha:
-            if image.channels >= 3:
+            if image.channels == 4:
+              // 4th bit is alpha -- if it's 0, skip drawing
+              if ((char*)src)[4*i+3] != 0:
```

Comment:
should it be i*image.channels? (i know image.channels == 4 so its guaranteed already, but just wondering)

Reply:
actually, maybe 4 == image.channels is coincidence with 4 bytes in a 32bit val

Reply:
yeah, image.channels is probably better. I think I was trying to squeeze this into one line at one point, and 4 was a lot shorter :) like `if src[i] != alpha || (image.channels == 4 && ((char*)src)[4*i+3] != 0)` but that condition felt kind of unwieldy, so I split it out.

Reply:
> actually, maybe 4 == image.channels is coincidence with 4 bytes in a 32bit val

oh that's interesting, maybe. I tried doing this w/ a 2-channel image as well and kept running into problems, but it might be b/c each pixel is represented as 32 bits.